cli/list/list.go

```diff
@@ -0,0 +1,161 @@
+package list
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/fatih/color"
+	"github.com/tarantool/tt/cli/cmdcontext"
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/util"
+)
+
+// printBinaries outputs installed versions of the program.
+func printBinaries(program string, binList []string, hasMaster bool, binActive string) {
+	if len(binList) > 0 || hasMaster {
+		log.Infof(program + ":")
+		if hasMaster {
+			if binActive == "master" {
+				fmt.Printf("	%s\n", util.Bold(color.GreenString("master - active")))
+			} else {
+				fmt.Printf("	%s\n", color.YellowString("master"))
+			}
+
+		}
+		for _, bin := range binList {
+			if bin == binActive {
+				fmt.Printf("	%s\n", util.Bold(color.GreenString(bin+" - active")))
+			} else {
+				fmt.Printf("	%s\n", color.YellowString(bin))
+			}
+		}
+	}
+}
+
+// sortBinaries sorts versions of binaries.
+func sortBinaries(binList []string) {
+	sort.SliceStable(binList, func(i, j int) bool {
+		varLeftString := strings.Split(binList[i], ".")
+		varRightString := strings.Split(binList[j], ".")
+		var right []uint64
+		var left []uint64
+
+		for _, ch := range varLeftString {
+			i, _ := strconv.Atoi(ch)
+			left = append(left, uint64(i))
+		}
+
+		for _, ch := range varRightString {
+			i, _ := strconv.Atoi(ch)
+			right = append(right, uint64(i))
+		}
+
+		largestLen := util.Max(len(left), len(right))
+
+		for i := 0; i < largestLen; i++ {
+			var valLeft, valRight uint64 = 0, 0
+			if i < len(left) {
+				valLeft = left[i]
+			}
+
+			if i < len(right) {
+				valRight = right[i]
+			}
+
+			if valLeft != valRight {
+				return valLeft > valRight
+			}
+		}
+
+		return false
+	})
+}
+
+// ListBinaries outputs installed versions of programs from bin_dir.
+func ListBinaries(cmdCtx *cmdcontext.CmdCtx, cliOpts *config.CliOpts) error {
+	var err error
+	if cliOpts.Repo == nil {
+		cliOpts.Repo = &config.RepoOpts{Install: "", Rocks: ""}
+	}
+	localDir := cliOpts.App.BinDir
+	if localDir == "" {
+		configDir := filepath.Dir(cmdCtx.Cli.ConfigPath)
+		localDir = filepath.Join(configDir, "bin")
+	}
+
+	localFiles, err := os.ReadDir(localDir)
+	if err != nil {
+		return err
+	}
+	var tarantoolBinaries []string
```

Comment:
The code below must be extracted to function, because it performs the same actions for tt & tarantool. This function must be testable. Current function must call it for tt and tarantool. And print the lists.

Reply:
Extracted